src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets

```diff
@@ -29,31 +29,57 @@ Copyright (c) .NET Foundation. All rights reserved.
     included in the generated deps.json.
     ============================================================
     -->
+  <UsingTask TaskName="FindNativeDeps" AssemblyFile="$(ILLinkTasksAssembly)" />
   <Target Name="_ILLink"
           Condition=" '$(PublishTrimmed)' == 'true' And
                       '$(_TargetFrameworkVersionWithoutV)' >= '3.0' And '$(TargetFrameworkIdentifier)' == '.NETCoreApp' "
           DependsOnTargets="_RunILLink">
 
+    <ItemGroup>
+      <_LinkedManagedAssemblies Include="@(_LinkedManagedAssembliesCandidates)" Condition="Exists('%(Identity)')" />
+    </ItemGroup>
+
+    <!-- Filter out native assets no longer needed after linking. -->
+    <ItemGroup Condition=" '$(TrimmerKeepNativeAssets)' != 'true' ">
+      <_NativeAssetsToAlwaysKeep Include="coreclr.dll;libcoreclr.dylib;libcoreclr.so" />
+      <_NativeAssetsToAlwaysKeep Include="clrjit.dll;libclrjit.dylib;libclrjit.so" />
+      <_NativeAssetsToAlwaysKeep Include="hostfxr.dll;libhostfxr.dylib;libhostfxr.so" />
+      <_NativeAssetsToAlwaysKeep Include="hostpolicy.dll;libhostpolicy.dylib;libhostpolicy.so" />
+      <!-- The full publish deps file path is generated after this target, so just depend on the filename. -->
+      <_NativeAssetsToAlwaysKeep Include="$(ProjectDepsFileName)" />
+      <_NativeAssetsToAlwaysKeep Include="$(ProjectRuntimeConfigFilePath)" />
```

Comment:
If the csproj exclicitly included a file to be published, I think we should not filter it away.
For example
```
<content include="libnativebinary.so">
     <CopyToPublishDirectory>Always</CopyToPublishDirectory?
</content>
```

Reply:
I believe that if the managed dlls use the native lib via a pinvoke, it will be kept due to a moduleref, but I need to check this. Maybe in addition we could also include a way to explicitly ask it to be kept, for cases where the native binary is used indirectly.

Reply:
Managed assemblies can use `NativeLibrary.Load()` methods to load native library from any string. This is similar to reflection in managed land. 

So, if the developer explicitly wanted something in the publish directory, I think we should leave it be.  

Is there any other way to tell the SDK to not remove a native binary?